qmlti/src/lti/model/student.php

```diff
@@ -232,65 +292,116 @@ function getAssessment() {
 
 /**
  * Gets number of attempts previously taken.
- * 
+ *
  * @return Integer Past attempts.
  */
   function getAttemptDetails() {
     if (!isset($_SESSION['error'])) {
-      $this->past_attempts = get_past_attempts($this->db, $this->resource_link_id, $this->assessment_id, $this->username);
+      if ($this->hasAttemptInProgress()) { # Already has an attempt setup
+        $this->past_attempts++;
+      }
     }
     return $this->past_attempts;
   }
 
 /**
+ * Gets past attempts list
+ *
+ * @return Student
+ */
+  function getPastAttempts() {
+    if (!isset($_SESSION['error'])) {
+      $this->past_attempts = get_past_attempts($this->db, $this->resource_link_id, $this->assessment_id, $this->username);
+    }
+    return $this;
+  }
+
+/**
  * Gets number of attempts available for assessment.
- * 
+ *
  * @return Integer Number of attempts.
  */
   function getNumberAttempts() {
     return $this->number_attempts;
   }
 
 /**
+ * Checks if user has attempt in progress.
+ *
+ * @return Boolean
+ */
+  function hasAttemptInProgress() {
+    return (get_latest_attempt($this->db, $this->consumer_key, $this->resource_link_id, $this->assessment_id, $this->username) != false);
+  }
+
+/**
  * Checks whether or not launch is disabled due to maximum attempts taken.
- * 
+ *
  * @return String UI disable string.
  */
   function checkLaunchDisabled() {
     if ($this->number_attempts != 'none') {
-      if ($this->past_attempts >= $this->number_attempts) {
-        return 'disabled';
-      } else {
+      if (($this->past_attempts >= $this->number_attempts) && (!$this->hasAttemptInProgress)) {
         return '';
+      } else {
+        return '<input class="btn btn-sm" type="submit" name="action" value="Launch Assessment" <?php echo $launch_disabled; />';
```

Comment:
Should you be doing '<input class="btn btn-sm" type="submit" name="action" value="Launch Assessment"' . $launch_disabled . '/>' here? Looks a bit weird echoing a variable in the middle of a return statement.